File: 2692-take-gifts-from-the-richest-pile/2692-take-gifts-from-the-richest-pile.cpp
class Solution {
public:
    long long pickGifts(vector<int>& gifts, int k) {
        for (int i = 1; i <= k; i++) {
            long long max = *max_element(gifts.begin(), gifts.end());
            for (int z=0;z<gifts.size();z++){
                if(gifts[z]==max){
                    gifts[z]=sqrt(max);
                    break;
                }
            }
        }
        long long x=0;
        for (int i=0;i<gifts.size();i++){
            x+=gifts[i];
        }
        return x;
    }
};

File: 2692-take-gifts-from-the-richest-pile/2692-take-gifts-from-the-richest-pile.py
class Solution:
    def pickGifts(self, gifts: List[int], k: int) -> int:
        while(k>0):
            r=max(gifts)
            for i in range(len(gifts)):
                if(r==gifts[i]):
                    gifts[i]=int(r**0.5)
                    k-=1
                    break
        
        return sum(gifts)

        